cluster/terraform/tls.tf

```diff
@@ -0,0 +1,31 @@
+data "azurerm_key_vault" "cert_kv" {
+  name                = var.cluster_kv
+  resource_group_name = var.resource_group_name
+}
+
+data "azurerm_key_vault_certificate_data" "cert" {
+  name         = local.cluster_cert_secret
+  key_vault_id = data.azurerm_key_vault.cert_kv.id
+}
+
+locals {
+  # The cert from the KV contains the full chain and is in the wrong order
+  # So we need to extract the primary cert which is the last in the list
+  full_cert = data.azurerm_key_vault_certificate_data.cert.pem
+  primary_cert = replace(replace(regexall("-----BEGIN CERTIFICATE-----[a-zA-Z0-9=+/]*-----END CERTIFICATE-----",
```

Comment:
Can we split this for readability? Something like:

```
cert_one_line = local.full_cert, "\n", "")
primary_cert_one_line = regexall("-----BEGIN CERTIFICATE-----[^-]*-----END CERTIFICATE-----", local.cert_one_line)[2]

primary_cert = replace(
  replace(
    local.primary_cert_one_line,
    "CERTIFICATE-----", "CERTIFICATE-----\n"
  ),
  "-----END", "\n-----END"
)
```

Reply:
done